core/src/main/scala/org/apache/spark/network/BlockTransferService.scala

```diff
@@ -26,24 +25,20 @@ import scala.reflect.ClassTag
 
 import org.apache.spark.internal.Logging
 import org.apache.spark.network.buffer.{FileSegmentManagedBuffer, ManagedBuffer, NioManagedBuffer}
-import org.apache.spark.network.shuffle.{BlockFetchingListener, DownloadFileManager, ShuffleClient}
+import org.apache.spark.network.shuffle.{BlockFetchingListener, BlockStoreClient, DownloadFileManager}
 import org.apache.spark.storage.{BlockId, EncryptedManagedBuffer, StorageLevel}
 import org.apache.spark.util.ThreadUtils
 
 private[spark]
-abstract class BlockTransferService extends ShuffleClient with Closeable with Logging {
+abstract class BlockTransferService extends BlockStoreClient with Logging {
```

Comment:
shall we add classdoc and mention that this class contains both server and client?

Reply:
Copy that, done in ae64538.